core/src/main/java/io/questdb/cutlass/parquet/CopyExportRequestJob.java

```diff
@@ -0,0 +1,244 @@
+    private final CairoEngine engine;
+    private final RingQueue<CopyExportRequestTask> requestQueue;
+    private final Sequence requestSubSeq;
+    private final SqlExecutionContextImpl sqlExecutionContext;
+    private final TableToken statusTableToken;
+    private final Utf8StringSink utf8StringSink = new Utf8StringSink();
+    private SerialParquetExporter serialExporter;
+    private TableWriter writer;
+
+    public CopyExportRequestJob(final CairoEngine engine) throws SqlException {
+        try {
+            this.requestQueue = engine.getMessageBus().getCopyExportRequestQueue();
+            this.requestSubSeq = engine.getMessageBus().getCopyExportRequestSubSeq();
+            this.serialExporter = new SerialParquetExporter(engine);
+
+            CairoConfiguration configuration = engine.getConfiguration();
+            this.clock = configuration.getMicrosecondClock();
+
+            final String statusTableName = configuration.getSystemTableNamePrefix() + "copy_export_log";
+            int logRetentionDays = configuration.getSqlCopyLogRetentionDays();
+            sqlExecutionContext = new SqlExecutionContextImpl(engine, 1);
+            try (SqlCompiler compiler = engine.getSqlCompiler()) {
+                sqlExecutionContext.with(configuration.getFactoryProvider().getSecurityContextFactory().getRootContext(), null, null);
+                this.statusTableToken = compiler.query()
+                        .$("CREATE TABLE IF NOT EXISTS \"")
+                        .$(statusTableName)
+                        .$("\" (" +
+                                "ts TIMESTAMP, " + // 0
+                                "id VARCHAR, " + // 1
+                                "table_name SYMBOL, " + // 2
+                                "files VARCHAR, " + // 3
+                                "phase SYMBOL, " + // 4
+                                "status SYMBOL, " + // 5
+                                "message VARCHAR, " + // 6
+                                "errors LONG" + // 7
+                                ") timestamp(ts) PARTITION BY DAY\n" +
+                                "TTL " + logRetentionDays + " DAYS BYPASS WAL;"
+                        )
+                        .createTable(sqlExecutionContext);
+            }
+
+            this.writer = engine.getWriter(statusTableToken, "QuestDB system");
+            this.copyContext = engine.getCopyExportContext();
+            this.copyContext.setReporter(this::updateStatus);
+            this.engine = engine;
+        } catch (Throwable t) {
+            close();
+            throw t;
+        }
+    }
+
+    @Override
+    public void close() {
+        this.serialExporter = Misc.free(serialExporter);
+        this.writer = Misc.free(this.writer);
+    }
+
+    public void updateStatus(
+            CopyExportRequestTask.Phase phase,
+            CopyExportRequestTask.Status status,
+            CopyExportRequestTask task,
+            Utf8Sequence files,
```

Comment:
In the right conditions, this string could overflow the column capacity. It can also take up lots of space in the log table.

Where is this string created? Do we check for the maximum size there?

I'm also unsure of its usefulness vs. space considerations.

Reply:
Yes, I didn't see where we create this value, but at least we must check for overflow.

Reply:
This is already the relative path to export dire, my opition is this would helpful for users to know how many files have been exported in this specific task. (The export folder may contain many files from different export tasks).

WDYT?

Reply:
We currently have `select * from export_files`, but it still display relative paths. I'm not sure if display absolute paths is a good idea.

Reply:
I cannot find this place anymore. We don't insert concatenated file names anymore